qiskit/transpiler/passes/swap_mapper.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+TODO
+"""
+
+from qiskit.transpiler._basepasses import TransformationPass
+from qiskit.dagcircuit import DAGCircuit
+
+
+class SwapMapper(TransformationPass):
+    """
+    Maps a DAGCircuit onto a `coupling_map` using swap gates.
+    """
+
+    def __init__(self, coupling_map, swap_basis_element='swap', swap_data=None):
+        """
+        Maps a DAGCircuit onto a `coupling_map` using swap gates.
+        Args:
+            coupling_map (Coupling): Directed graph represented a coupling map.
+            swap_basis_element (string):  Default: 'swap' the name of the gate
+               that will be used for swaping.
+            swap_data (dict): The swap "gate data". Default: the swap gate is opaque.
```

Comment:
About the arguments to the SwapMapper's `__init__`:
- Why does it need a "swap_basis_element" and "swap_data"? Under what circumstances would I pass something other than the default here?

- Can we also make it take an `initial_layout` as well? We will soon call this a `desired_layout`. But basically, it's a just a layout that says if my program complies with this, don't change it.

Reply:
There are many talking points here. So, moving the the conversation to https://github.com/Qiskit/qiskit-terra/pull/1270#issuecomment-439951585